tests/test-utils.ts

```diff
@@ -1,12 +1,71 @@
-import jest from 'jest-mock';
+import { jest } from '@jest/globals';
 
-/**
- * Mock the fetch function
- * @param {*} status
- * @param {*} payload
- * @return {Object}
- */
-export function mockFetch(status, payload) {
+interface FetchResponse {
+  json: () => Promise<any>;
+  text: () => Promise<string>;
+  status: number;
+  ok: boolean;
+}
+
+interface FetchStreamResponse {
+  status: number;
+  ok: boolean;
+  body: AsyncGenerator;
+}
+
+interface ModelPermission {
+  id: string;
+  object: string;
+  created: number;
+  allow_create_engine: boolean;
+  allow_sampling: boolean;
+  allow_logprobs: boolean;
+  allow_search_indices: boolean;
+  allow_view: boolean;
+  allow_fine_tuning: boolean;
+  organization: string;
+  group: null | string;
+  is_blocking: boolean;
+}
+
+interface Model {
+  id: string;
+  object: string;
+  created: number;
+  owned_by: string;
+  root: null;
+  parent: null;
+  permission: ModelPermission[];
+}
+
+interface ModelsList {
+  object: string;
+  data: Model[];
+}
+
+interface ChatCompletionChoice {
+  finish_reason: string | null;
+  message?: {
+    role: string;
+    content: string;
+  };
+  index: number;
+}
+
+interface ChatCompletion {
+  id: string;
+  object: string;
+  created: number;
+  choices: ChatCompletionChoice[];
+  model: string;
+  usage: {
+    prompt_tokens: number;
+    total_tokens: number;
+    completion_tokens: number;
+  };
+}
```

Comment:
Why do you need to put the interfaces again here?

Reply:
just clean it in further commit.